integration-tests/src/test/java/uk/gov/di/authentication/shared/services/AuthOrchSerializationServicesIntegrationTest.java

```diff
@@ -1,37 +1,150 @@
+                        authConfigurationService, authRedisConnectionService);
+    }
+
+    @Test
+    void authCanReadFromSessionCreatedByOrch() {
+        var orchSession = orchSessionService.generateSession();
+        var sessionId = orchSession.getSessionId();
+        orchSession.addClientSession(CLIENT_SESSION_ID);
+        orchSessionService.storeOrUpdateSession(orchSession);
+        var authSession = authSessionService.getSession(sessionId).get();
+        assertThat(authSession.getClientSessions(), contains(CLIENT_SESSION_ID));
+    }
+
+    @Test
+    void orchCanReadFromSessionCreatedByAuth() {
+        var sessionId = "some-existing-session-id";
+        var authSession = new Session(sessionId);
+        authSession.addClientSession(CLIENT_SESSION_ID);
+        authSessionService.storeOrUpdateSession(authSession);
+        var orchSession = orchSessionService.getSession(sessionId).get();
+        assertThat(orchSession.getClientSessions(), contains(CLIENT_SESSION_ID));
+    }
+
+    @Test
+    void authCanUpdateSharedFieldInSessionCreatedByOrch() {
+        var orchSession = orchSessionService.generateSession();
+        var sessionId = orchSession.getSessionId();
+        orchSessionService.storeOrUpdateSession(orchSession);
+        var authSession = authSessionService.getSession(sessionId).get();
+        authSession.addClientSession(CLIENT_SESSION_ID);
+        authSessionService.storeOrUpdateSession(authSession);
+        orchSession = orchSessionService.getSession(sessionId).get();
+        assertThat(orchSession.getClientSessions(), contains(CLIENT_SESSION_ID));
+    }
+
+    @Test
+    void orchCanUpdateSharedFieldInSessionCreatedByAuth() {
+        var sessionId = "some-existing-session-id";
+        var authSession = new Session(sessionId);
+        authSessionService.storeOrUpdateSession(authSession);
+        var orchSession = orchSessionService.getSession(sessionId).get();
+        orchSession.addClientSession(CLIENT_SESSION_ID);
+        orchSessionService.storeOrUpdateSession(orchSession);
+        authSession = authSessionService.getSession(sessionId).get();
+        assertThat(authSession.getClientSessions(), contains(CLIENT_SESSION_ID));
+    }
+
+    @Test
+    void orchCanReadUnsharedFieldAfterAuthUpdatesSession() {
+        var orchSession = orchSessionService.generateSession();
+        var sessionId = orchSession.getSessionId();
+        orchSession.setBrowserSessionId(BROWSER_SESSION_ID);
+        orchSessionService.storeOrUpdateSession(orchSession);
+        var authSession = authSessionService.getSession(sessionId).get();
+        authSession.addClientSession(CLIENT_SESSION_ID);
+        authSessionService.storeOrUpdateSession(authSession);
+        orchSession = orchSessionService.getSession(sessionId).get();
+        assertThat(
+                orchSession.getBrowserSessionId(), is(equalTo(orchSession.getBrowserSessionId())));
```

Comment:
I think one of these should be replaced with `BROWSER_SESSION_ID`. Currently asserting that it is itself

Reply:
Haha, probably